tests/integration/helm/helm_upgrade_test.go

```diff
@@ -0,0 +1,132 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"path/filepath"
+	"testing"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/helm"
+)
+
+var (
+	// previousChartPath is path of Helm charts for previous Istio deployments.
+	previousChartPath = filepath.Join(env.IstioSrc, "tests/integration/helm/testdata/")
+)
+
+const (
+	gcrHub                   = "gcr.io/istio-release"
+	previousSupportedVersion = "1.8.1"
+
+	defaultValues = `
+global:
+  hub: %s
+  tag: %s
+`
+
+	firstPartyJwtValues = `
+global:
+  hub: %s
+  tag: %s
+  jwtPolicy: first-party-jwt
+`
+)
+
+// TestDefaultInPlaceUpgrades tests Istio installation using Helm with default options
+func TestDefaultInPlaceUpgrades(t *testing.T) {
+	framework.
+		NewTest(t).
+		Features("installation.helm.default.upgrade").
+		Run(func(ctx framework.TestContext) {
+			installThenUpgrade(t, ctx, defaultValues)
+		})
+
+}
+
+// TestInstallWithFirstPartyJwt tests Istio installation using Helm
+// with first-party-jwt enabled
+func TestUpgradesWithFirstPartyJwt(t *testing.T) {
+	framework.
```

Comment:
Why do we care about jwt policy in upgrade? This type of testing has  a massive risk of combinatorial explosion since there's near infinite number of combinations of settings, starting and ending versions. Jwt policy, to me, seems like one of the least interesting values (I felt the same for initial install, but less big of a deal there as it was just one test) - there is effectively 0 chance of this test failing when the default one passes and vice versa. And even if it did provide additional coverage this is for legacy insecure code - anyone using this is "doing it wrong" and should fix their k8s install.

Sorry for the rant :slightly_smiling_face: let me know if there is extra coverage we are getting by this. Would be good to look at how long each additional test takes as well

Reply:
It was already present, so I added an upgrade test. The de facto standard should be to support third-party-jwt so I'm fine with removing this test.

But that got me thinking that I should modify the test to do a N-1 proxy to N proxy test to provide more value. I've changed it to do so.

Reply:
On my local cluster the tests take ~1 minute or so each. With the way they are run in integration testing with kind and side loading the images it should be faster, assuming there's enough cpu/memory to be scheduled appropriately.